packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -340,6 +355,67 @@ export function executeStepDependencyGraph<
       enqueueLeafSteps();
     }
 
+    /**
+     * Loads cached step data.
+     * @param stepCacheFilePath
+     * @param context
+     */
+    async function loadCacheForStep(
+      stepCacheFilePath: string,
+      context: TStepExecutionContext,
+    ) {
+      let status = StepResultStatus.FAILURE;
+      const entitiesPath = `${stepCacheFilePath}/entities`;
+      const relationshipsPath = `${stepCacheFilePath}/relationships`;
+
+      const { jobState, logger } = context;
+
+      let entitiesCount = 0;
+      await walkDirectory({
```

Comment:
We should be able to replace this with `iterateParsedGraphFiles` which is exported from `packages/integration-sdk-runtime/src/fileSystem.ts`.

Cleanup suggestion...

There is a function called `iterateParsedGraphFiles`, which is exported from `packages/integration-sdk-runtime/src/fileSystem.ts`. I think we could expose two new functions:

- `iteratedParsedEntityGraphFiles`
- `iteratedParsedRelationshipGraphFiles`

Example impl:

```ts
export function iteratedParsedEntityGraphFiles(
  iteratee: (entities: Entity[]) => Promise<void>,
  graphPath?: string
) {
  return iterateParsedGraphFiles(async (data) => {
    if (data.entities) await iteratee(data.entities);
  }, graphPath);
}
```

Example usage:

```ts
await iteratedParsedEntityGraphFiles(
  async (entities) => {
    entitiesCount += entities.length;
    await jobState.addEntities(entities);
    status = StepResultStatus.CACHED;
  },
  entitiesPath
);
```